apps/src/templates/sectionProgress/progressTables/progressTableHelpers.js

```diff
@@ -1,14 +1,119 @@
+import React from 'react';
 import moment from 'moment';
+import {lessonIsAllAssessment} from '@cdo/apps/templates/progress/progressHelpers';
+import ProgressTableSummaryCell from './ProgressTableSummaryCell';
+import ProgressTableDetailCell from './ProgressTableDetailCell';
+import ProgressTableLevelSpacer from './ProgressTableLevelSpacer';
+import ProgressTableLevelIconSet from './ProgressTableLevelIconSet';
+import * as progressStyles from '@cdo/apps/templates/progress/progressStyles';
 
-export function timeSpentFormatter(studentProgress) {
+/**
+ * @return {Array} Array of formatter functions for the progress table summary view
+ * mainCellFormatter is used to format the main cell (summary of progress on lesson) for each student
+ * timeSpentCellFormatter is used to format the time spent cell (time spent on the lesson) in the expanded view
+ * lastUpdatedCellFormatter is used to format the last updated cell (progress last updated) in the expanded view
+ *
+ * @param {Object} lessonProgressByStudent
+ * An object mapping student id to an object with lesson id mapping to studentLessonProgressType
+ * @param {function} onClickLesson
+ * A function which is called when a summary cell is clicked
+ *
+ * */
+export function getSummaryCellFormatters(
```

Comment:
nice encapsulation here. seems like a pretty clean solution to a pretty dirty problem. :)